File: backend/src/constants/TurnaroundStatus.go
package constants

// TurnaroundStatus values, duplicated in frontend constants/TurnaroundStatus.
const (
	StatusArriving  = "ARRIVING"
	StatusOnStand   = "ON_STAND"
	StatusInService = "IN_SERVICE"
	StatusReady     = "READY"
	StatusDeparted  = "DEPARTED"
	StatusDelayed   = "DELAYED"
)

var TurnaroundStatus = []string{
	StatusArriving,
	StatusOnStand,
	StatusInService,
	StatusReady,
	StatusDeparted,
	StatusDelayed,
}

var TurnaroundStatusText = map[string]string{
	StatusArriving:  "即将到站",
	StatusOnStand:   "已靠桥",
	StatusInService: "保障中",
	StatusReady:     "待放行",
	StatusDeparted:  "已离港",
	StatusDelayed:   "延误",
}

func IsValidTurnaroundStatus(value string) bool {
	for _, item := range TurnaroundStatus {
		if item == value {
			return true
		}
	}
	return false
}

File: frontend/src/types/TurnaroundStatus.ts
export type TurnaroundStatus =
  | "ARRIVING"
  | "ON_STAND"
  | "IN_SERVICE"
  | "READY"
  | "DEPARTED"
  | "DELAYED";

export const TURNAROUND_STATUSES: TurnaroundStatus[] = [
  "ARRIVING",
  "ON_STAND",
  "IN_SERVICE",
  "READY",
  "DEPARTED",
  "DELAYED",
];

export const TURNAROUND_STATUS_TEXT: Record<TurnaroundStatus, string> = {
  ARRIVING: "即将到站",
  ON_STAND: "已靠桥",
  IN_SERVICE: "保障中",
  READY: "待放行",
  DEPARTED: "已离港",
  DELAYED: "延误",
};
